crates/rome_js_analyze/src/analyzers/nursery/no_misused_new.rs

```diff
@@ -0,0 +1,232 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::{markup, MarkupBuf};
+use rome_js_syntax::{
+    AnyTsTypeMember, JsClassDeclaration, JsLanguage, TsDeclareStatement, TsInterfaceDeclaration,
+    TsReferenceType, TsTypeAliasDeclaration,
+};
+use rome_rowan::{declare_node_union, AstNode, SyntaxToken, TextRange};
+
+declare_rule! {
+    /// Enforce proper usage of `new` and `constructor`.
+    ///
+    /// In JavaScript, classes utilize the `constructor` method to initialize a new instance. On the other hand, TypeScript interfaces can describe a class type with a `new()` method signature, though this pattern is not commonly seen in real-world code. Developers, especially those new to JavaScript or TypeScript, might occasionally confuse the use of `constructor` with `new`.
+    /// This rule triggers warnings in the following scenarios:
+    /// - When a class has a method named `new`.
+    /// - When an interface defines a method named `constructor` or `new` that returns the interface type.
+    /// - When a type alias has a `constructor` method.
+    ///
+    /// You should not use this rule if you intentionally want a class with a `new` method, and you're confident nobody working in your code will mistake it with an `constructor`.
+    ///
+    /// Source: https://typescript-eslint.io/rules/no-misused-new/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// declare class C {
+    ///   new(): C;
+    /// }
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// interface I {
+    ///   new (): I;
+    ///   constructor(): void;
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```ts
+    /// declare class C {
+    ///   constructor();
+    /// }
+    ///
+    /// interface I {
+    ///   new (): C;
+    /// }
+    /// ```
+    ///
+    pub(crate) NoMisusedNew {
+        version: "next",
+        name: "noMisusedNew",
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub(crate) DeclarationQuery = TsInterfaceDeclaration |  TsTypeAliasDeclaration | JsClassDeclaration | TsDeclareStatement
+}
+
+pub(crate) enum RuleState {
+    ClassMisusedNew(TextRange),
+    InterfaceMisusedNew(TextRange),
+    InterfaceMisusedConstructor(TextRange),
+    TypeAliasMisusedConstructor(TextRange),
+}
+
+impl RuleState {
+    fn message(&self) -> MarkupBuf {
+        match self {
+            RuleState::ClassMisusedNew(_) => (markup! {
+                "Do not use the `new` method in class."
```

Comment:
You can emphase `new` and `constructor`:

```suggestion
                "Do not use the `"<Emphasis>"new"</Emphasis>"` method in class."
```

Maybe we could use a contracted form of `do not`: `don't`. Not sure about that, we should check what other rules are using.

Reply:
A query `"Do not` hits 12 results and a query `Don't` hits 8 results in `biome_js_analyze/**/*.rs`.
 
`Do not` may be harsh. If we unify the form, I also like `Don't`.